src/lib.rs

```diff
@@ -27,6 +27,39 @@
 //! rely on file paths for _some_ operations. See the security documentation on
 //! the `NamedTempFile` type for more information.
 //!
+//! ## Early drop pitfall
+//! Because `TempDir` and `NamedTempFile` rely on their destructors for cleanup, this can lead
+//! to an unexpected early removal of the directory/file, usually when working with APIs which are
+//! generic over `AsRef<Path>`. Consider the following example:
+//!
+//! ```
```

Comment:
I'm not sure if this example should be kept as a doc test or maybe markes as `no_run` since it would fail on platforms where touch is not available.

Reply:
Let's mark it `no_run` for now.